pkg/sql/delegate/show_functions.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package delegate
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/lexbase"
+	"github.com/cockroachdb/cockroach/pkg/sql/opt/cat"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/catconstants"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+)
+
+// delegateShowFunctions implements SHOW FUNCTIONS which returns all the
+// user-defined functions.
+// Privileges: None.
+//
+//	Notes: postgres does not have a SHOW FUNCTIONS statement.
+func (d *delegator) delegateShowFunctions(n *tree.ShowFunctions) (tree.Statement, error) {
+	flags := cat.Flags{AvoidDescriptorCaches: true}
+	_, name, err := d.catalog.ResolveSchema(d.ctx, flags, &n.ObjectNamePrefix)
+	if err != nil {
+		return nil, err
+	}
+	// If we're resolved a one-part name into <db>.public (which is the behavior
+	// of ResolveSchema, not for any obviously good reason), rework the resolved
+	// name to have an explicit catalog but no explicit schema. This would arise
+	// when doing SHOW FUNCTIONS FROM <db>. Without this logic, we would not show the
+	// tables from other schemas than public.
+	if name.ExplicitSchema && name.ExplicitCatalog && name.SchemaName == tree.PublicSchemaName &&
+		n.ExplicitSchema && !n.ExplicitCatalog && n.SchemaName == name.CatalogName {
+		name.SchemaName, name.ExplicitSchema = "", false
+	}
+	var schemaClause string
+	if name.ExplicitSchema {
+		schema := lexbase.EscapeSQLString(name.Schema())
```

Comment:
hmm, could you explain why do we need to escape the resolved names here?

Reply:
`escapeSQLString` also takes care of adding single-quotes around the `go` string, so therefore it becomes safe to use in the `AND n.nspame` clause below.